flytekit/tools/lazy_loader.py

```diff
@@ -26,12 +26,16 @@ def get_extras_require(cls):
         """
         d = cls.LAZY_LOADING_PLUGINS.copy()
         all_plugins = []
+        all_plugins3 = []
         for k in d:
-            # Default to Spark 2.4.x .
+            # Default to Spark 2.4.x in all and Spark 3.x in all3.
             if k != "spark3":
                 all_plugins.extend(d[k])
+            if k != "spark":
+                all_plugins3.extend(d[k])
 
         d["all"] = all_plugins
+        d["all3"] = all_plugins3
```

Comment:
I actually think that `all` should be spark3 and `all-spark2.4` should be spark2.
Let us atleast not call it all3 -> `all-spark3`

Reply:
We lose backward-compatibility at that point. I have cut an issue to move `all` to spark3 at some point in the future. Can rename all3 to be more verbose.

Reply:
Renamed